csrc/executor.cpp

```diff
@@ -1606,6 +1607,137 @@ void FusionExecutor::initializeExecutorEntry(
   executor_entry.init = true;
 }
 
+/// Copies the data, logical_size, and alloc_stride parameters to the
+/// appropriate parts of entry.args[idx].
+///
+/// For GPU tensors, we pass a Tensor<type, rank, rank> struct (see
+/// runtime/tensor.cu), where the rank describes the number of elements in the
+/// shape and stride arrays. The actual shapes and strides are dynamic, but the
+/// type and rank of the tensors are actually static (changing them would need
+/// a new FusionDefinition). So we create the storage area for the
+/// Tensor<t,r,r> during ::computeArgs, and then in this function we just
+/// update that memory with the current values for the tensor's base address,
+/// shape, and strides.
+///
+/// @param entry the entry we have previously setup for this fusion
+/// @param idx the index into entry.args and related parallel arrays in the
+///            entry.
+/// @param idx_type_size generally sizeof(int32_t) or sizeof(int64_t); used for
+///                      computing how large the arrays to copy are.
+static void fill_gpu_ptr(
```

Comment:
Kernels can actually take CPU tensors as well, at least according to getKA and specifically polymorphicValueToBytes.

Granted the only CPU "tensors" supported are scalars, but nonetheless...

`TensorMetaData` is a real object in our system (actually something that's used inside this function), and so I hesitate to use that name because that's not what's happening here. Maybe `tensorArgMD`?

Reply:
You mean something like fillTensorArgMetadata? Yeah, LGTM!